### PR TITLE
added the ability to manage orders

### PR DESCRIPTION
**Pull request by AndreyPolezhaiev**

### CODE REVIEW

File: src/main/java/com/polezhaiev/shop/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            Authentication authentication,
+            @PathVariable Long orderId,
+            @RequestBody OrderUpdateStatusRequestDto requestDto) {
+        User user = (User) authentication.getPrincipal();
+        return orderService.updateStatus(orderId, requestDto);
+    }
+
+    @Operation(summary = "Delete the item", description = "Delete the item")
+    @DeleteMapping("/{id}")
+    public void deleteOrderItemById(@PathVariable Long id) {
```
Comment (sokimaaa) [inline src/main/java/com/polezhaiev/shop/controller/OrderController.java]: hasRole user? 

Only registered user should have ability to delete own orderItem (nor ANY user can edit order of user with id=1)

File: src/main/java/com/polezhaiev/shop/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @Operation(summary = "Place the order", description = "Place the order")
+    @PostMapping
+    public OrderResponseDto placeOrder(
+            Authentication authentication, @RequestBody OrderMakeRequestDto requestDto) {
```
Comment (sokimaaa) [inline src/main/java/com/polezhaiev/shop/controller/OrderController.java]: validation?

File: src/main/java/com/polezhaiev/shop/dto/order/order/OrderMakeRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.polezhaiev.shop.dto.order.order;
+
+import lombok.Data;
+
+@Data
+public class OrderMakeRequestDto {
+    private String shippingAddress;
```
Comment (sokimaaa) [inline src/main/java/com/polezhaiev/shop/dto/order/order/OrderMakeRequestDto.java:9]: consider adding validation whenever needed.

File: src/main/java/com/polezhaiev/shop/controller/OrderController.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/com/polezhaiev/shop/controller/OrderController.java]: not fixed

File: src/main/java/com/polezhaiev/shop/dto/order/order/OrderMakeRequestDto.java (hunk 3 above)
Comment (koval-kn) [reply src/main/java/com/polezhaiev/shop/dto/order/order/OrderMakeRequestDto.java:9]: not fixed

File: src/main/java/com/polezhaiev/shop/dto/order/order/OrderUpdateStatusRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.polezhaiev.shop.dto.order.order;
+
+import com.polezhaiev.shop.model.Status;
+import lombok.Data;
+
+@Data
+public class OrderUpdateStatusRequestDto {
+    private Status status;
```
Comment (koval-kn) [inline src/main/java/com/polezhaiev/shop/dto/order/order/OrderUpdateStatusRequestDto.java:10]: add validation

File: src/main/java/com/polezhaiev/shop/service/order/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        return orderMapper.toDto(updated);
+    }
+
+    @Override
+    public void deleteOrderItemById(Long userId, Long id) {
+        Long userIdFromRepo = orderItemRepository.getById(id).getOrder().getUser().getId();
+        if (userId.equals(userIdFromRepo)) {
+            orderItemRepository.deleteById(id);
+        } else {
+            throw new NoSuchElementException(
```
Comment (koval-kn) [inline src/main/java/com/polezhaiev/shop/service/order/impl/OrderServiceImpl.java]: create your own exception with name related to this situation
```suggestion
            throw new AccessDeniedException(
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

**Pull request merged**